user_manual/changes/process_changes.py

```diff
@@ -139,56 +141,53 @@ def get_entries(*args):
 
 def rst(entries, args):
     """
-    Print RST for all entries, structured by change type, with headers for each
-    change type.
+    Print reST for all entries, structured by change type, with headers for
+    each change type.
     """
     buffr = []
 
-    for entry in entries:
-        entry['rst'] = entry2rst(entry, args.show_date)
-
-    if not args.quiet:
-        for change_type in types_to_header.keys():
-            header_chunk = types_to_header[change_type]
-
-            filtered_entries = sorted(
-                (e for e in entries if e['type'] == change_type),
-                key=lambda e: e['date'], reverse=True
-            )
+    # Generate one big reST document to hold all entries, grouped by change
+    # type.
+    for change_type in types_to_header.keys():
+        # Collect all entries for this change type, more recent first
+        filtered_entries = sorted(
+            (e for e in entries if e['type'] == change_type),
+            key=lambda e: e['date'], reverse=True
+        )
 
-            if filtered_entries == []:
-                continue
+        # No need to emit a section for this change type if it has no entry
+        if filtered_entries == []:
+            continue
 
-            buffr.append(
-                header('Libadalang API {}'.format(header_chunk), '#')
-            )
-
-            for entry in filtered_entries:
-                buffr.append(entry['rst'])
+        header_chunk = types_to_header[change_type]
+        buffr.append(
+            header('Libadalang API {}'.format(header_chunk), '#')
+        )
+        for entry in filtered_entries:
+            buffr.append(entry2rst(entry, args.show_date))
 
     result = '\n'.join(buffr)
 
-    # We always generate the html, even if we don't need a preview, so that
-    # we check that the rst is correctly formatted.
+    # We always generate the HTML, even if we don't need a preview, so that we
+    # check that the reST is correctly formatted.
     html = publish_string(result, writer_name='html')
+    if not args.quiet:
+        print(result)
+        print(html)
```

Comment:
It's still wrong (and it's my fault): we're not supposed to print html, that's a remain from my own debugging ... Documentation generation will fail this way.

Reply:
Good point, thanks! I’ve removed the extra `print` statement.